src/extensions/default/JavaScriptCodeHints/main.js

```diff
@@ -859,6 +859,62 @@ define(function (require, exports, module) {
         }
 
         /*
+         * Handle Rename menu/keyboard command.
+         */
+        function handleRename() {
+            var offset,
+                handleFindRefs;
+
+
+            // Only provide rename when cursor is in JavaScript content
+            if (!session || session.editor.getModeForSelection() !== "javascript") {
```

Comment:
Can `session.editor` be `null` or `undefined`? Most likely not, but might be worth checking out.